relay-general/src/store/transactions.rs

```diff
@@ -203,6 +213,54 @@ fn set_default_transaction_source(event: &mut Event) {
     }
 }
 
+/// Normalize the transaction name.
+///
+/// Replaces UUIDs, SHAs and numerical IDs in transaction names by placeholders.
+fn normalize_transaction_name(transaction: &mut Annotated<String>) -> ProcessingResult {
+    let capture_names = TRANSACTION_NAME_NORMALIZER_REGEX
+        .capture_names()
+        .flatten()
+        .collect::<Vec<_>>();
+
+    transaction.apply(|trans, meta| {
+        // Collect all the remarks first, since we will mutate the string afterwords and all the
+        // ranges will change.
+        for matches in TRANSACTION_NAME_NORMALIZER_REGEX.captures_iter(trans) {
+            // Skip first, since it's always the entire match and the following are groups.
+            // And if there is no matches, just cotinue to next iteration.
+            let mut matches_iter = matches.iter();
+            if matches_iter.next().is_none() {
+                continue;
+            }
+            for m in matches_iter.flatten() {
+                let remark = Remark::with_range(
+                    RemarkType::Substituted,
+                    "normalize_transaction_name",
+                    (m.start(), m.end()),
+                );
+                meta.add_remark(remark)
+            }
+        }
+        for name in capture_names {
+            if let Some(caps) = TRANSACTION_NAME_NORMALIZER_REGEX.captures(trans) {
+                if let Some(m) = caps.name(name) {
+                    if meta.original_value().is_none() {
+                        meta.set_original_value(Some(trans.to_string()))
+                    }
+                    let mut collector = String::new();
```

Comment:
It should be possible to allocate a new String only once for each transaction name. Assuming that `capture_names` are ordered , we can start writing `[0..m1.start()]`, then `[m1.end()...m2.start()]`, etc.